### PR TITLE
Added linting with GitHub workflow

### LINKED ISSUE #15: Adding GitHub actions Linting
**Is your feature request related to a problem? Please describe.**

We need to add tests for GitHub Actions to check all PRs.

**Describe the solution you'd like**

We can start this with properly configuring ESLint, such that we add another script to the project, which we then run using GitHub actions.

**Describe alternatives you've considered**

We can use either GitHub actions or stuff like Travis CI.

### PR DESCRIPTION
**Pull request by ritvij14**
**Description**

**Related Issue** 
- issue goes here

Fixes #15 

- [x] Code
- [x] Testing

**Code/Quality Assurance Only**

- This change requires a documentation update (software upgrade on readme file)

**Additional Info (OPTIONAL)**

- Properly added eslint related scripts to `package.json`
- Added a GitHub Workflow with tests for any linting errors, to maintain code style of the project
- Formatted the code according to the linting rules.

**Checklist**

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by ritvij14** [comment]
Please review this PR and tell if any changes to the linting rules are needed.

**Comment by Abhishek-kumar09** [comment]
Please show the results by running github actions on your own branch.

**Comment by ritvij14** [comment]
@Abhishek-kumar09 here
the commit -: https://github.com/ritvij14/edu-server/commit/bf8a5eeee1ab5095f7763ce4a8c2b4f7d1667e86

the test run -: https://github.com/ritvij14/edu-server/runs/2107783046

**Comment by Abhishek-kumar09** [comment]
It has warnings and still the actions is showing green tick.
https://github.com/ritvij14/edu-server/runs/2107783046?check_suite_focus=true

Linting action should fail when you get errors or warnings. Set the default severity to errors for github action linting.

**Comment by ritvij14** [comment]
@Abhishek-kumar09 could you please tell me how should I be setting the default severity in GitHub workflow? For now, I have added "error" to the no-unused-vars rule in eslint.

**Comment by Abhishek-kumar09** [comment]
The dedfault severity is error by default. Can you share the github actions with the latest changes.
Sorry for late comment, kind of having some deadline issues with other projects.

**Comment by ritvij14** [comment]
@Abhishek-kumar09 it's now counting the previous warning as an error too
https://github.com/ritvij14/edu-server/runs/2130589559#step:4:29

The thing was that that particular warning was being counted as a warning locally too. And it doesn't fail tests for warnings from eslint, only errors. Hence it was passing last time.

File: package-lock.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "project",
+  "name": "edu-client",
```
Comment (Abhishek-kumar09) [inline package-lock.json:2]: Change it to edu-server

**Pull request merged**